File: DS/Graph.java
package CS-520-EleNA-System.DS;

import java.util.ArrayList;
import java.util.HashMap;

public class Graph {
    // This HashMap maps a string representing the name of the place to a node containing its information.
    private HashMap<String, Node> nameNodeMap;
    // This HashMap maps a node to the edges leaving from that node.
    private HashMap<Node, ArrayList<Edge>> nodeEdgeMap;

    public Graph() {
        super();
    }

    public void mapStringToNode(String name, Node node) {
        this.nameNodeMap.put(name, node);
    }

    public void mapNodeToEdge(Node node, Edge edge) {
        if (this.nodeEdgeMap.containsKey(node)) {
            this.nodeEdgeMap.get(node).add(edge);
        }
        else {
            this.nodeEdgeMap.put(node, new ArrayList<Edge>());
            this.nodeEdgeMap.get(node).add(edge);
        }
    }

    public HashMap<String, Node> getNameNodeMap() {
        return this.nameNodeMap;
    }

    public HashMap<Node, Edge> getNodeEdgeMap() {
        return this.nodeEdgeMap;
    }

    
    
}

File: DS/Graph.py
class Graph:
    # This HashMap maps a string representing the name of the place to a node containing its information.
    # private HashMap<String, Node> nameNodeMap;
    # This HashMap maps a node to the edges leaving from that node.
    # private HashMap<Node, ArrayList<Edge>> nodeEdgeMap;

    def __init__(self):
        self._idMap = {}
        # self._nodeEdgeMap = {}
    

    def mapIdToNode(self, id, node):
        self._nameNodeMap[id] = node

    # def mapNodeToEdge(self, node, edge):
    #     if (node in self._nodeEdgeMap):
    #         self._nodeEdgeMap.get(node).append(edge)
    #     else:
    #         self._nodeEdgeMap[node] = []
    #         self._nodeEdgeMap.get(node).append(edge)
    

    def getNodeMap(self):
        return self._nameNodeMap

    # def getNodeEdgeMap(self):
    #     return self._nodeEdgeMap
    